library/pod-security-policy/users/template.yaml

```diff
@@ -15,55 +15,87 @@ spec:
           properties:
             runAsUser:
               type: object
+              description: "Controls which user ID pod containers can be run with."
               properties:
                 rule:
                   type: string
+                  description: "A strategy for applying the runAsUser restriction."
+                  enum:
+                    - MustRunAs
+                    - MustRunAsNonRoot
+                    - RunAsAny
                 ranges:
                   type: array
+                  description: "A list of ranges affected by the rule."
                   items:
                     type: object
+                    description: "The range of user IDs affected by the rule."
                     properties:
                       min:
                         type: integer
                       max:
                         type: integer
             runAsGroup:
               type: object
+              description: "Controls which group ID pod containers can be run with."
               properties:
                 rule:
                   type: string
+                  description: "A strategy for applying the runAsGroup restriction."
+                  enum:
+                    - MustRunAs
+                    - MayRunAs
+                    - RunAsAny
                 ranges:
                   type: array
+                  description: "A list of ranges affected by the rule."
                   items:
                     type: object
+                    description: "The range of group IDs affected by the rule."
                     properties:
                       min:
                         type: integer
                       max:
                         type: integer
             supplementalGroups:
               type: object
+              description: "Controls which groups are applied to the first process run in any pod container, in addition to the container's primary GID."
               properties:
                 rule:
                   type: string
+                  description: "A strategy for applying the supplementalGroups restriction."
+                  enum:
+                    - MustRunAs
+                    - MayRunAs
+                    - RunAsAny
                 ranges:
                   type: array
+                  description: "A list of ranges affected by the rule."
                   items:
                     type: object
+                    description: "The range of group IDs affected by the rule."
                     properties:
                       min:
                         type: integer
                       max:
                         type: integer
             fsGroup:
               type: object
+              description: "Controls the supplemental group applied to some volumes."
```

Comment:
"some volumes": which volumes?

Reply:
Also [stolen from k8s docs](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems:~:text=FSGroup%20%2D%20Controls%20the%20supplemental%20group,any%20fsGroup%20ID%20to%20be%20specified.).

Based on [this blogpost](https://medium.com/kubernetes-tutorials/defining-privileges-and-access-control-settings-for-pods-and-containers-in-kubernetes-2cef08fc62b7#d6f3:~:text=.spec.securityContext.fsGroup%20%E2%80%94%20The%20field%20defines%20a,owning%20GID%20will%20be%20the%20fsGroup.) and [these docs](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#pods-security-security-context-yaml:~:text=Since%20fsGroup%20field%20is%20specified%2C%20all,volume%20will%20be%20Group%20ID%202000.), it looks like it sets both the volume mounted to the contianer AND any files created within it by the Pod containers have this GID.

This is complicated, and I'm not sure how to think about ROI on figuring out how it works in detail.  Is there something that's kind of vague but not wrong that we could say?  It seems like the k8s docs punted as well

Reply:
Please reword anything inspired from others' work.

Reply:
Also, we should be documenting these things based of the Rego written, for accuracy.

Finally, I'm asking "which volumes", because as a user, that would be my first question when I read something like "some volumes.

Reply:
I think you make a good point in [another comment](https://github.com/open-policy-agent/gatekeeper-library/pull/109#discussion_r712661299) that we shouldn't try to teach people these features, but rather how to restrict them via the CT.

That being said, I think it makes sense to change the descriptions in this CT to things like:

```yaml
fsGroup:
  type: object
  description: "Controls the fsGroup values allowed in a Pod or container level SecurityContext."
```

This is simpler and is actually what we validate in the rego.

Going to go ahead and change these, let me know if I should revert.